### PR TITLE
feat: add mergify integration

### PR DESCRIPTION
**Pull request by sileht**
### What does this PR do?

This change introduces the boilerplate and first metric of Mergify integration

### Motivation

We want to make Mergify <-> DataDog integration easier to install and more visible.

### Review checklist

- [X] PR has a [meaningful title](https://github.com/DataDog/integrations-core/blob/master/CONTRIBUTING.md#pull-request-title) or PR has the `no-changelog` label attached
- [X] Feature or bugfix has tests
- [X] Git history is clean
- [ ] If PR impacts documentation, docs team has been notified or an issue has been opened on the [documentation repo](https://github.com/DataDog/documentation/issues/new)

### Additional Notes

Anything else we should know when reviewing?

### CODE REVIEW

**Comment by kayayarai** [comment]
Created DOCS-5059 to track docs review

File: mergify/README.md [hunk 1]
```diff
@@ -0,0 +1,61 @@
+# Agent Check: Mergify
+
+## Overview
+
+This check tracks [Mergify][1] merge-queue statistics.
```
Comment (jd) [inline mergify/README.md:5]:
```suggestion
This check tracks [Mergify][1] merge queue statistics.
```

File: mergify/README.md [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+Need help? Contact [Mergify support][1].
+
+[1]: https://mergify.com
+[2]: https://app.datadoghq.com/account/settings#agent
+[3]: https://github.com/DataDog/integrations-extras/blob/master/mergify/datadog_checks/mergify/data/conf.yaml.example
+[4]: https://docs.datadoghq.com/agent/guide/agent-commands/#start-stop-and-restart-the-agent
+[5]: https://docs.datadoghq.com/agent/guide/agent-commands/#agent-status-and-information
+[6]: https://github.com/DataDog/integrations-extras/blob/master/mergify/metadata.csv
+[7]: https://github.com/DataDog/integrations-extras/blob/master/mergify/assets/service_checks.json
+
```
Comment (buraizu) [inline mergify/README.md:61]:
```suggestion
[8]: https://docs.datadoghq.com/developers/integrations/new_check_howto/#developer-toolkit
```

File: mergify/README.md [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+## Setup
+
+### Installation
+
+To install the Mergify check on your host:
+
+
+1. Install the [developer toolkit]
+(https://docs.datadoghq.com/developers/integrations/new_check_howto/#developer-toolkit)
+ on any machine.
```
Comment (buraizu) [inline mergify/README.md:16]:
```suggestion

1. Install the [developer toolkit][8] on any machine.
```

File: mergify/README.md [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+
+3. [Download the Datadog Agent][2].
+
+4. Upload the build artifact to any host with an Agent and
+ run `datadog-agent integration install -w
+ path/to/mergify/dist/<ARTIFACT_NAME>.whl`.
+
+### Configuration
+
+1. Edit the `mergify.d/conf.yaml` file in the `conf.d/` folder at the root of your [Agent's configuration directory][3] to start collecting your Mergify [metrics](#metrics). See the [
```
Comment (buraizu) [inline mergify/README.md:28]: Looks like this step may still be under development -- could you please confirm and update?

File: mergify/metadata.csv [hunk 5]
```diff
@@ -0,0 +1,2 @@
+metric_name,metric_type,interval,unit_name,per_unit_name,description,orientation,integration,short_name,curated_metric
+mergify.merge_queue_length,gauge,,,,merge queue length,0,mergify,,
```
Comment (buraizu) [inline mergify/metadata.csv:2]:
```suggestion
mergify.merge_queue_length,gauge,,,,Merge queue length,0,mergify,,
```

**Review by buraizu — CHANGES_REQUESTED** [review]
Thanks for the PR, I've left just a few minor edits to the docs and flagged a section that looks to be still under development.

File: mergify/README.md (hunk 3 above)
Comment (sileht) [reply mergify/README.md]: Fixed.

File: mergify/README.md (hunk 2 above)
Comment (sileht) [reply mergify/README.md]: Fixed.

File: mergify/README.md (hunk 4 above)
Comment (sileht) [reply mergify/README.md]: I re-added the missing sentence.

File: mergify/metadata.csv (hunk 5 above)
Comment (sileht) [reply mergify/metadata.csv]: Fixed.

File: mergify/README.md [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+
+4. Upload the build artifact to any host with an Agent and
+ run `datadog-agent integration install -w
+ path/to/mergify/dist/<ARTIFACT_NAME>.whl`.
+
+### Configuration
+
+1. Edit the `mergify.d/conf.yaml` file in the `conf.d/` folder at the root of your [Agent's configuration directory][11] to start collecting your Mergify [metrics](#metrics).
+
+See the sample [mergify.d/conf.yaml.example][3] file for all available configuration options.
```
Comment (buraizu) [inline mergify/README.md:27]:
```suggestion
   See the sample [mergify.d/conf.yaml.example][3] file for all available configuration options.
```

**Review by buraizu — CHANGES_REQUESTED** [review]
Thanks for implementing those changes, just made a few additional suggestions after reviewing a preview build of the page.

**Review by buraizu — APPROVED** [review]
Thanks for implementing those changes, this looks good from a documentation perspective

**Comment by sileht** [comment]
I just rebased the PR to fix the conflict.

File: mergify/tests/conftest.py [hunk 7]
```diff
@@ -0,0 +1,46 @@
...
+    return {
+        "mergify_api_url": "https://mergify.example.com",
+        "repositories": {"owner/repository": ["main"]},
+        "token": "a_token",
+        "tags": ["test:mergify"],
+    }
+
+
+@pytest.fixture()
+def mergify_api_queues():
```
Comment (iliakur) [inline mergify/tests/conftest.py:34]: Not a blocker, but we also have a fixture for this, [here's an example of how to use it](https://github.com/DataDog/integrations-core/blob/a8c41fb2312bc17c2e1255d3e23d3a92776cea2c/rabbitmq/tests/test_openmetrics.py#L57).

File: mergify/datadog_checks/mergify/check.py [hunk 8]
```diff
@@ -0,0 +1,75 @@
...
+
+    def check(self, _):
+        # type: (Any) -> None
+        headers = {
+            "Authorization": "Bearer %s" % self.token,
+            "Content-Type": "application/json",
+        }
+
+        for repository, branches in self.repositories.items():
+            queue_url = "%s/v1/repos/%s/queues" % (self.api_url, repository)
```
Comment (iliakur) [inline mergify/datadog_checks/mergify/check.py:31]: Let's use `str.format` or f-strings here please.

**Review by iliakur — CHANGES_REQUESTED** [review]
Apologies for the delayed response. Just a couple minor comments.

**Review by iliakur — APPROVED** [review]

**Review by jd — APPROVED** [review]

**Comment by iliakur** [comment]
@sileht could you please remove the unused imports? I'm sorry that they got caught only now, my bad. Once that's done, feel free to tag me in a comment and I'll merge the PR.

**Comment by sileht** [comment]
> @sileht could you please remove the unused imports? I'm sorry that they got caught only now, my bad. Once that's done, feel free to tag me in a comment and I'll merge the PR.

@iliakur  Done

File: .github/CODEOWNERS [hunk 9]
```diff
@@ -362,8 +363,14 @@
 /logzio/*metadata.csv                    @DataDog/agent-integrations @DataDog/documentation
 /logzio/manifest.json                    @DataDog/agent-integrations @DataDog/documentation
 /logzio/README.md                        @DataDog/agent-integrations @DataDog/documentation
+<<<<<<< HEAD
```
Comment (bgoldberg122) [inline .github/CODEOWNERS:366]: @iliakur there were merge conflicts in this file -- Do we need all 5 of these lines?

**Pull request merged**